src/NexusMods.App.UI/LeftMenu/Items/ApplyControlViewModel.cs

```diff
@@ -40,56 +49,72 @@ public ApplyControlViewModel(LoadoutId loadoutId, IServiceProvider serviceProvid
         _applyService = serviceProvider.GetRequiredService<IApplyService>();
         LaunchButtonViewModel = serviceProvider.GetRequiredService<ILaunchButtonViewModel>();
         LaunchButtonViewModel.LoadoutId = loadoutId;
+        
+        NewestLoadout = _loadoutRegistry.Get(_loadoutId) ??
+                       throw new ArgumentException("Loadout not found: " + _loadoutId);
+        
+        _gameInstallation = NewestLoadout.Installation;
 
-
-        _gameInstallation = _loadoutRegistry.Get(_loadoutId)?.Installation ??
-                            throw new ArgumentException("Loadout not found: " + _loadoutId);
-
-        (_lastAppliedLoadoutId, _lastAppliedRevisionId) = GetLastAppliedLoadout();
-
-        _newestLoadout = _loadoutRegistry.RevisionsAsLoadouts(loadoutId)
-            .ToProperty(this, vm => vm.NewestLoadout, scheduler: RxApp.MainThreadScheduler);
+        LastAppliedRevisionId = _applyService.GetLastAppliedLoadout(_gameInstallation) ??
+                                throw new ArgumentException("No last applied loadout found for: " +
+                                                            _gameInstallation);
         
         _applyReactiveCommand = ReactiveCommand.CreateFromTask(async () => await Apply());
 
         this.WhenActivated(disposables =>
         {
+            _loadoutRegistry.RevisionsAsLoadouts(loadoutId)
+                .OnUI()
+                .BindToVM(this, vm => vm.NewestLoadout)
+                .DisposeWith(disposables);
+
+            _applyService.LastAppliedRevisionFor(_gameInstallation)
+                .OnUI()
+                .BindToVM(this, vm => vm.LastAppliedRevisionId)
+                .DisposeWith(disposables);
+            
+            this.WhenAnyValue(vm => vm.LastAppliedRevisionId)
+                .Select(revId =>
+                {
+                    var loadout = _loadoutRegistry.GetLoadout(revId);
+                    if (loadout is null)
+                        throw new ArgumentException("Loadout not found for revision: " + revId);
+                    return loadout.LoadoutId;
+                })
+                .OnUI()
+                .BindToVM(this, vm => vm.LastAppliedLoadoutId)
+                .DisposeWith(disposables);
```

Comment:
`LastAppliedLoadoutId` is set by two different observable streams, you should merge these two streams into one.

Reply:
One is last applied `RevisionId`, one is last applied `LoadoutId`

Reply:
Ah nvm.